src/util/merge.js

```diff
@@ -0,0 +1,71 @@
+var hasOwnProperty = Object.prototype.hasOwnProperty;
```

Comment:
Going to trust you on this one for now :) 

Is this copied/pasted, or written from scratch?